src/plugins/backend/backend.c

```diff
@@ -0,0 +1,700 @@
+						 "The plugin referenced in '%s%s' (value: '%s', refering to '%s/plugins/%s') could not be "
+						 "found. (Configuration of mountpoint: '%s')",
+						 keyName (parentKey), keyName (pluginRefKey), pluginRef, keyName (parentKey), pluginRef,
+						 keyBaseName (parentKey));
+		return false;
+	}
+
+	if (type == PLUGIN_TYPE_GET && (*pluginPtr)->kdbGet == NULL)
+	{
+		ELEKTRA_SET_INSTALLATION_ERRORF (parentKey,
+						 "The plugin '%s' was referenced in a kdbGet() position ('%s%s'), but does not implement "
+						 "kdbGet(). (Configuration of mountpoint: '%s')",
+						 (*pluginPtr)->name, keyName (parentKey), keyName (pluginRefKey), keyBaseName (parentKey));
+		return false;
+	}
+
+	if (type == PLUGIN_TYPE_SET && (*pluginPtr)->kdbSet == NULL)
+	{
+		ELEKTRA_SET_INSTALLATION_ERRORF (parentKey,
+						 "The plugin '%s' was referenced in a kdbSet() position ('%s%s'), but does not implement "
+						 "kdbSet(). (Configuration of mountpoint: '%s')",
+						 (*pluginPtr)->name, keyName (parentKey), keyName (pluginRefKey), keyBaseName (parentKey));
+		return false;
+	}
+
+	if (type == PLUGIN_TYPE_COMMIT && (*pluginPtr)->kdbCommit == NULL)
+	{
+		ELEKTRA_SET_INSTALLATION_ERRORF (parentKey,
+						 "The plugin '%s' was referenced in a kdbCommit() position ('%s%s'), but does not "
+						 "implement kdbCommit(). (Configuration of mountpoint: '%s')",
+						 (*pluginPtr)->name, keyName (parentKey), keyName (pluginRefKey), keyBaseName (parentKey));
+		return false;
+	}
+
+	if (type == PLUGIN_TYPE_ERROR && (*pluginPtr)->kdbError == NULL)
+	{
+		ELEKTRA_SET_INSTALLATION_ERRORF (parentKey,
+						 "The plugin '%s' was referenced in a kdbError() position ('%s%s'), but does not implement "
+						 "kdbError(). (Configuration of mountpoint: '%s')",
+						 (*pluginPtr)->name, keyName (parentKey), keyName (pluginRefKey), keyBaseName (parentKey));
+		return false;
+	}
+
+	return true;
+}
+
+static bool loadPluginList (PluginList ** pluginListPtr, Plugin * thisPlugin, KeySet * definition, const char * pluginRefRootName,
+			    enum PluginType type, Key * parentKey)
+{
+	Key * pluginRefRoot = keyNew (pluginRefRootName, KEY_END);
+
+	*pluginListPtr = NULL;
+	PluginList * listEnd = NULL;
+
+	for (elektraCursor end, i = ksFindHierarchy (definition, pluginRefRoot, &end); i < end; i++)
+	{
+		Key * cur = ksAtCursor (definition, i);
+		if (keyIsDirectlyBelow (pluginRefRoot, cur) != 1)
+		{
+			continue;
+		}
+
+		Plugin * plugin;
+		if (!loadPlugin (&plugin, thisPlugin, cur, type, parentKey))
+		{
+			keyDel (pluginRefRoot);
+			return false;
+		}
+
+		PluginList * element = elektraMalloc (sizeof (PluginList));
+		element->plugin = plugin;
+		element->next = NULL;
+
+		if (listEnd == NULL)
+		{
+			*pluginListPtr = element;
+		}
+		else
+		{
+			listEnd->next = element;
+		}
+		listEnd = element;
+	}
+
+	keyDel (pluginRefRoot);
+	return true;
+}
+
+int ELEKTRA_PLUGIN_FUNCTION (init) (Plugin * plugin, KeySet * definition, Key * parentKey)
```

Comment:
I thought init is only a phase, not a function? This should also be discussed together with commit&error.

Reply:
I did make it a function originally, but now I believe we should use the `get` function and make it only a phase.

Reply:
I fully agree! :rocket: